packages/libp2p/src/circuit-relay/transport/index.ts

```diff
@@ -133,6 +135,13 @@ class CircuitRelayTransport implements Transport {
     this.maxOutboundStopStreams = init.maxOutboundStopStreams ?? defaults.maxOutboundStopStreams
     this.stopTimeout = init.stopTimeout ?? defaults.stopTimeout
 
+    this.shutdownController = new AbortController()
```

Comment:
This controller seems to be created only so we can increase the max number of listeners on it's signal.

The signal is not passed into any other context that might actually add some event listeners to it - how does this solved the problem?

Reply:
Right, good catch. It turns out it was actually the transport listener not removing the [relayStore event listener](https://github.com/libp2p/js-libp2p/blob/9a69e6f76018d2df5c6e51de930557d4b60b2fca/packages/libp2p/src/circuit-relay/transport/listener.ts#L34) when closing the connection as we discussed.

I decided against adding a test as the [relay spec already tests for connection close events](https://github.com/libp2p/js-libp2p/blob/9a69e6f76018d2df5c6e51de930557d4b60b2fca/packages/libp2p/test/circuit-relay/relay.spec.ts#L103) and so given the connection is closed via the [safeDispatch](https://github.com/libp2p/js-libp2p/blob/9a69e6f76018d2df5c6e51de930557d4b60b2fca/packages/libp2p/src/circuit-relay/transport/listener.ts#L104) then the `close` method is called (which then emits the close connection event). 

Seperately I can see the event being removed in the logs
<img width="621" alt="image" src="https://github.com/libp2p/js-libp2p/assets/9755286/fcd499f5-c747-4bb0-a120-2b17f4e3a5bc">